### ISSUE TITLE
Improve selected area visuals when resizing a column or a row

### ISSUE THREAD

**Issue by llorca** [palantir/blueprint#843]
What it looks like now:
![image](https://cloud.githubusercontent.com/assets/199754/23970654/c6c00afe-0987-11e7-9223-1d55bb5909fc.png)

Do we want to show the selection rectangle too?

**Comment by tgreenwatts** [comment]
I actually don't mind this selection behavior personally

**Comment by llorca** [comment]
Yeah, the interaction seems fine but I wonder what @pkwi thinks for the visuals -- it's pretty muddy at the moment.

Also note that Excel does the opposite: on column resize, they strip column header styles and keep the selection rectangle around

**Comment by pkwi** [comment]
To start with - do we need that extra blue divider between selected columns?

<img width="391" alt="screen shot 2017-03-16 at 11 57 24 am" src="https://cloud.githubusercontent.com/assets/23030180/23995094/e88e5da8-0a3f-11e7-8ec2-518c92408a8b.png">

We could remove it and keep the blue stroke only around the whole selection like:

<img width="361" alt="screen shot 2017-03-16 at 11 57 33 am" src="https://cloud.githubusercontent.com/assets/23030180/23995101/f8e08834-0a3f-11e7-9610-620aa46ee738.png">

Also, the blue line used for resizing feels very heavy and chunky, could we not use a dashed or dotted, thinner line?

I think interaction is fine, however it would be great to see the blue selection while I resize the column - as @llorca said: similar to what Excel does.

**Comment by gscshoyru** [comment]
@pkwi the "extra" blue line you're talking about should only show up if you select the columns as separate selection regions. If you select multiple columns at once, by dragging, then there won't be that extra line. 

I think we want to keep the extra line to show them as separate regions -- I know excel doesn't do that, but google spreadsheets does, and I prefer showing something to users as opposed to not, here.

**Comment by pkwi** [comment]
gotcha! I see that the same goes for cells. That makes sense.

Sidenote: another interaction I noticed is missing it's the ability to select a range of cells by clicking on the first cell and shift + click on the last cell.

**Comment by gscshoyru** [comment]
Yeah, that one is tracked here: https://github.com/palantir/blueprint/issues/823

**Comment by llorca** [comment]
@pkwi thanks for the comments! We're tracking those things. Back to the original topic, how can we improve this muddy blue-ish, gray-ish column header when resizing occurring?

**Comment by pkwi** [comment]
The muddy blue comes from the mixture of semi-transparent blue and grey background. I'm not sure you can do much here to visually improve it, without changing the actual colour (replace transparency with a solid colour). Or you can drop the selection / hover state while resizing columns.

**Comment by llorca** [comment]
Got it... So to summarize, either we drop all the selection visuals on column/row resize, or either we keep them all. No in-between like currently

**Comment by pkwi** [comment]
@llorca yep. Not sure which one is better visually tbh. Would probably have to try it out. I guess keeping all selection makes more sense as you won't confuse users who might think they've lost their selections.

**Comment by llorca** [comment]
Good point. Let's see if it's possible to try out both in order to get a better idea of the interaction here

**Comment by tgreenwatts** [comment]
What's the action remaining here?

**Comment by cmslewis** [comment]
// Reanimating this thread

@tgreenwatts @llorca @pkwi One minimal remaining action here is to fix the header-cell hover states while resizing is in progress. Notice all the flickering on `mouseenter`/`mouseleave`

![2017-10-11 11 03 55](https://user-images.githubusercontent.com/443450/31458145-f6d338c4-ae73-11e7-97cd-2e2ace3ee0d2.gif)
